lntest/itest/lnd_psbt_test.go

```diff
@@ -470,6 +472,193 @@ func testPsbtChanFundingExternal(net *lntest.NetworkHarness, t *harnessTest) {
 	closeChannelAndAssert(t, net, carol, chanPoint2, false)
 }
 
+// testPsbtChanFundingSingleStep checks whether PSBT funding works also when the
+// wallet of both nodes are empty and one of them uses PSBT and an external
+// wallet to fund the channel while creating reserve output in the same
+// transaction.
+func testPsbtChanFundingSingleStep(net *lntest.NetworkHarness, t *harnessTest) {
+	ctxb := context.Background()
+	const chanSize = funding.MaxBtcFundingAmount
+
+	args := nodeArgsForCommitType(lnrpc.CommitmentType_ANCHORS)
+	// First, we'll create two new nodes that we'll use to open channels
+	// between for this test. But in this case both nodes have an empty
+	// wallet.
+	carol := net.NewNode(t.t, "carol", args)
+	defer shutdownAndAssert(net, t, carol)
+
+	dave := net.NewNode(t.t, "dave", args)
+	defer shutdownAndAssert(net, t, dave)
+
+	// Third node is only used for its wallet with PSBT funding abilities
+	funder := net.NewNode(t.t, "funder", nil)
```

Comment:
Can we use `net.Alice` here instead? That node already has coins. This prevents us from creating too many new nodes (which always takes time and mines a bunch of blocks).